pkg/eval/mods/file/file.go

```diff
@@ -124,3 +127,43 @@ func prclose(p vals.Pipe) error {
 func pwclose(p vals.Pipe) error {
 	return p.WriteEnd.Close()
 }
+
+//elvdoc:fn truncate
+func truncate(name string, size vals.Num) error {
+
+	switch size.(type) {
+	case int:
+		s := int64(size.(int))
+		err := os.Truncate(name, s)
+		if err != nil {
+			return err
+		} else {
+			return nil
+		}
+
+	case *big.Int:
+		if (size.(*big.Int)).IsInt64() {
+			s := (size.(*big.Int)).Int64()
+			err := os.Truncate(name, s)
+			if err != nil {
+				return err
+			} else {
+				return nil
+			}
+		} else {
+			return errs.OutOfRange{
+				What:      "size argument for os.Truncate",
+				ValidLow:  "0",
+				ValidHigh: "8 bytes",
+				Actual:    "More than 8 bytes",
+			}
+		}
+
+	default:
+		return errs.BadValue{
+			What:   "size argument for os.Truncate",
+			Valid:  "int or *big.Int",
+			Actual: "float64",
```

Comment:
Don't hardcode the `"float64"` type string since this could be some other type.

@xiaq: It seems to me this function (and some of the code in the `math` module) should be using `ScanToGo()` rather than an explicit `switch size.(type)`. However, at the present time `ScanToGo` is not sufficiently general for and doesn't support commonly used Go types like `int64`.

Reply:
Noted, will fix this.